app/controllers/mixins/sandbox.rb

```diff
@@ -25,6 +25,7 @@ def x_tree_init(name, type, leaf)
   end
 
   def x_active_tree
+    @sb ||= {}
```

Comment:
maybe a DRYer way to do this is via a new `sandbox` method:
```ruby
def sandbox
  @sb ||= {}
end
```

then `x_active_tree` becomes
```ruby
def x_active_tree
  sandbox[:active_tree]
end
```